Userland/Applications/TextEditor/MainWidget.cpp

```diff
@@ -272,34 +274,41 @@ MainWidget::MainWidget()
                 return;
         }
 
-        open_file(open_path.value());
+        open_file(fd_response.fd().value().take_fd(), fd_response.chosen_file().value());
     });
 
     m_save_as_action = GUI::CommonActions::make_save_as_action([&](auto&) {
-        Optional<String> save_path = GUI::FilePicker::get_save_filepath(window(), m_name.is_null() ? "Untitled" : m_name, m_extension.is_null() ? "txt" : m_extension);
-        if (!save_path.has_value())
+        auto response = m_filesystem_client->prompt_save_file(m_name.is_null() ? "Untitled" : m_name, m_extension.is_null() ? "txt" : m_extension, Core::StandardPaths::home_directory(), FileSystemAccessServer::WriteCreateTruncate);
+
+        if (response.error() != 0)
             return;
 
-        if (!m_editor->write_to_file(save_path.value())) {
+        if (!m_editor->write_to_file(response.fd().value().take_fd())) {
```

Comment:
Just FYI, with `Optional`, you can use `*` and `->` to access the value at bit less verbose.
```cpp
response.fd()->take_fd()
```
It's not wrong how you did it, just as a heads-up.
If you just want the value of `response.fd()`, you could use `*response.fd()`.

Reply:
Ah very cool! I prefer this style, thanks.